src/pointer.rs

```diff
@@ -422,6 +422,53 @@ impl Pointer {
     pub fn components(&self) -> Components {
         self.into()
     }
+
+    /// Appends `token` onto `self` as a new [`Pointerbuf`].
+    ///
+    /// ## Example
+    /// ```
+    /// let ptr = jsonptr::Pointer::from_static("/foo");
+    /// let foobar = ptr.with_trailing_token("bar");
+    /// assert_eq!(foobar, "/foo/bar");
+    /// ```
+    pub fn with_trailing_token<'t, T>(&self, token: T) -> PointerBuf
+    where
+        T: Into<Token<'t>>,
```

Comment:
Minor nit, but I think `impl Into<Token<'_>>` would be cleaner (unsure about whether the lifetime can be elided but I think so)